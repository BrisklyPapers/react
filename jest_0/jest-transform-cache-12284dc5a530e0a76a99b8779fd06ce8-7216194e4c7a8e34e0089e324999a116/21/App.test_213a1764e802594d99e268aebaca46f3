'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _App = require('../../src/components/App.js');var _App2 = _interopRequireDefault(_App);

var _MuiThemeProvider = require('material-ui/styles/MuiThemeProvider');var _MuiThemeProvider2 = _interopRequireDefault(_MuiThemeProvider);
var _reactRedux = require('react-redux');
var _fetchMock = require('fetch-mock');var _fetchMock2 = _interopRequireDefault(_fetchMock);
var _reduxMockStore = require('redux-mock-store');var _reduxMockStore2 = _interopRequireDefault(_reduxMockStore);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}var ReactShallowRenderer = require('react-test-renderer/shallow');

var mockStore = (0, _reduxMockStore2.default)();
var renderer = new ReactShallowRenderer();

describe('components/App', function () {
    it('renders correctly', function () {
        var app = renderer.render(
        _react2.default.createElement(_App2.default, null));

        expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
});