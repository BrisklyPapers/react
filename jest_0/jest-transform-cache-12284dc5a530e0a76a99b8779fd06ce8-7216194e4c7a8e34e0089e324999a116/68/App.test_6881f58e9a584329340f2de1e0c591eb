'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _App = require('../../src/components/App.js');var _App2 = _interopRequireDefault(_App);
var _reactTestRenderer = require('react-test-renderer');var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('components/App', function () {
    it('renders correctly', function () {
        var app = _reactTestRenderer2.default.create(
        _react2.default.createElement(_App2.default, null));

        expect(app.toJSON()).toMatchSnapshot();
    });
});