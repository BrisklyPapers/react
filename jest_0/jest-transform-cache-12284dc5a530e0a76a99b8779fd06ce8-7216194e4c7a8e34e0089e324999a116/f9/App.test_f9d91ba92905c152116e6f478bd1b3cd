'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _App = require('../../src/components/App.js');var _App2 = _interopRequireDefault(_App);
var _reactTestRenderer = require('react-test-renderer');var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);
var _MuiThemeProvider = require('material-ui/styles/MuiThemeProvider');var _MuiThemeProvider2 = _interopRequireDefault(_MuiThemeProvider);
var _reactRedux = require('react-redux');
var _fetchMock = require('fetch-mock');var _fetchMock2 = _interopRequireDefault(_fetchMock);
var _reduxMockStore = require('redux-mock-store');var _reduxMockStore2 = _interopRequireDefault(_reduxMockStore);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

var mockStore = (0, _reduxMockStore2.default)();

describe('components/App', function () {
    it('renders correctly', function () {
        var app = _reactTestRenderer2.default.create(
        _react2.default.createElement(_MuiThemeProvider2.default, null,
            _react2.default.createElement(_reactRedux.Provider, { store: mockStore({ searchedText: "" }) },
                _react2.default.createElement(_App2.default, null))));



        expect(app.toJSON()).toMatchSnapshot();
    });
});