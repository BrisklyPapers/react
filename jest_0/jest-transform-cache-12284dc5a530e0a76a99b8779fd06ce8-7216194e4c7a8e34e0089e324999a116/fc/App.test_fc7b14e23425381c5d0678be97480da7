'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _App = require('../../src/components/App.js');var _App2 = _interopRequireDefault(_App);

var _MuiThemeProvider = require('material-ui/styles/MuiThemeProvider');var _MuiThemeProvider2 = _interopRequireDefault(_MuiThemeProvider);
var _reactRedux = require('react-redux');
var _fetchMock = require('fetch-mock');var _fetchMock2 = _interopRequireDefault(_fetchMock);
var _reduxMockStore = require('redux-mock-store');var _reduxMockStore2 = _interopRequireDefault(_reduxMockStore);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}var ReactShallowRenderer = require('react-test-renderer/shallow');

var mockStore = (0, _reduxMockStore2.default)();
var renderer = new ReactShallowRenderer();

describe('components/App', function () {
    it('renders correctly', function () {
        var app = renderer.render(
        _react2.default.createElement(_MuiThemeProvider2.default, null,
            _react2.default.createElement(_reactRedux.Provider, { store: mockStore({ searchedText: "" }) },
                _react2.default.createElement(_App2.default, null))));



        expect(app.getRenderOutput()).toMatchSnapshot();
    });
});