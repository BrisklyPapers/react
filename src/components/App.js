import React from 'react';
import Navigation from '../containers/Navigation';
import IndexPage from '../containers/IndexPage';
import ResultPage from '../containers/ResultPage';
import FileUploadPage from '../containers/FileUploadPage';
import SearchResults from '../containers/SearchResults';
import DropZone from '../containers/DropZone';
import IndexPageHead from './IndexPageHead';
import IndexPageDescription from './IndexPageDescription';
import Radium from 'radium';
import {connect} from 'react-redux';

class App extends React.Component {
    render() {
        return (
            <div>
                <div style={styles.divOnTop[this.props.cssStyle]}>
                    <Navigation />
                    <IndexPageHead />
                    <ResultPage>
                        <SearchResults />
                    </ResultPage>
                    <FileUploadPage>
                        <DropZone />
                    </FileUploadPage>
                </div>
                <IndexPageDescription />
            </div>
        )
    };
}

var styles = {
    divOnTop: {
        index: {
            backgroundImage: 'url("/home_background_1920cs.jpg")',
            height: "100vh",
            backgroundSize: 'cover',
            backgroundPosition: 'bottom center',
            backgroundRepeat: 'no-repeat',
            position: "relative"
        },
        login: {
            backgroundImage: 'url("/login_background_1920x420.jpg")',
            minHeight: "100vh",
            height: "100%",
            backgroundSize: '100% 1px',
            backgroundPosition: 'bottom center',
            backgroundRepeat: 'repeat-y',
            position: "relative"
        }
    }
};

export default Radium(App);