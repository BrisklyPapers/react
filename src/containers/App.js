import { connect } from 'react-redux';
import AppComponent from '../components/App';

const mapStateToProps = (state) => {
    return {
        loggedIn: state.logged_in,
        cssStyle: state.logged_in ? 'login' : 'index'
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(AppComponent);