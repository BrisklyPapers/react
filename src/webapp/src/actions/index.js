import fetch from 'isomorphic-fetch'

export const SEARCH_INPUT_KEY_PRESSED = 'SEARCH_INPUT_KEY_PRESSED';
export function searchInputKeyPressed(text) {
    return {
        type: SEARCH_INPUT_KEY_PRESSED,
        text
    }
}

export const SEARCH_DOCUMENTS = 'SEARCH_DOCUMENTS';
export function searchDocuments(text) {
    return {
        type: SEARCH_DOCUMENTS,
        text
    }
}

export const RECEIVE_DOCUMENTS = 'RECEIVE_DOCUMENTS';
function receiveDocuments(json) {
    console.log(json.data);
    return {
        type: RECEIVE_DOCUMENTS,
        documents: [], //json.data.children.map(child => child.data),
        receivedAt: Date.now()
    }
}

export const RECEIVE_DOCUMENTS_ERROR = 'RECEIVE_DOCUMENTS_ERROR';
function receiveDocumentsError(json) {
    return {
        type: RECEIVE_DOCUMENTS_ERROR,
        documents: [],
        receivedAt: Date.now()
    }
}

// store.dispatch(fetchPosts())
export function fetchDocuments(text) {
    // Thunk middleware knows how to handle functions.
    // It passes the dispatch method as an argument to the function,
    // thus making it able to dispatch actions itself.

    return function (dispatch) {
        // First dispatch: the app state is updated to inform
        // that the API call is starting.

        dispatch(searchDocuments(text))

        // The function called by the thunk middleware can return a value,
        // that is passed on as the return value of the dispatch method.

        // In this case, we return a promise to wait for.
        // This is not required by thunk middleware, but it is convenient for us.

        return fetch(`http://localhost:8085/document/search?q=${text}`)
            .then(
                response => response.json(),
                // Do not use catch, because that will also catch
                // any errors in the dispatch and resulting render,
                // causing an loop of 'Unexpected batch number' errors.
                // https://github.com/facebook/react/issues/6895
                error => console.log('An error occured.', error)
            )
            .then(json =>
                // We can dispatch many times!
                // Here, we update the app state with the results of the API call.

                dispatch(receiveDocuments(json))
            )
    }
}

export function fetchDocumentsIfNeeded(text) {
    // Note that the function also receives getState()
    // which lets you choose what to dispatch next.

    // This is useful for avoiding a network request if
    // a cached value is already available.

    return (dispatch, getState) => {
        if (1) {
            // Dispatch a thunk from thunk!
            return dispatch(fetchDocuments(text))
        } else {
            // Let the calling code know there's nothing to wait for.
            return Promise.resolve()
        }
    }
}