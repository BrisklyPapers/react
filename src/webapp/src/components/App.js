import React from 'react';
import Search from '../containers/Search';
import SearchResults from '../containers/SearchResults';

const App = () => (
    <div>
        <Search />
        <SearchResults />
    </div>
);

export default App;