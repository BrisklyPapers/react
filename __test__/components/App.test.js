import React from 'react';
import App from '../../src/components/App.js';
import {PAGEVIEW_INDEX, PAGEVIEW_SEARCH_RESULTS} from '../../src/actions';
const ReactShallowRenderer = require('react-test-renderer/shallow');
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { Provider } from 'react-redux';
import fetchMock from 'fetch-mock'
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();
const renderer = new ReactShallowRenderer();

describe('components/App', () => {
    it('renders correctly', () => {
        renderer.render(
            <App store={mockStore({loggedIn: false, navigation: PAGEVIEW_INDEX})} cssStyle="index" loggedIn={false}/>
        );
        expect(renderer.getRenderOutput()).toMatchSnapshot();
    });

    it('renders correctly when logged in', () => {
        renderer.render(
            <App store={mockStore({loggedIn: true, navigation: PAGEVIEW_SEARCH_RESULTS})} cssStyle="login" loggedIn={true}/>
        );
        expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
});