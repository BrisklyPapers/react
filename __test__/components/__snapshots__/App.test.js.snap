// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/App renders correctly 1`] = `
<div
  data-radium={true}
>
  <div
    data-radium={true}
    style={
      Object {
        "backgroundImage": "url(\\"/home_background_1920cs.jpg\\")",
        "backgroundPosition": "bottom center",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "cover",
        "height": "100vh",
        "position": "relative",
      }
    }
  >
    <Connect(Navigation) />
    <IndexPageHead />
    <Connect(PageView)>
      <Connect(SearchResults) />
    </Connect(PageView)>
    <Connect(PageView)>
      <Connect(DropZoneComponent) />
    </Connect(PageView)>
  </div>
  <IndexPageDescription />
</div>
`;

exports[`components/App renders correctly when logged in 1`] = `
<div
  data-radium={true}
>
  <div
    data-radium={true}
    style={
      Object {
        "backgroundImage": "url(\\"/login_background_1920x420.jpg\\")",
        "backgroundPosition": "bottom center",
        "backgroundRepeat": "repeat-y",
        "backgroundSize": "100% 1px",
        "height": "100%",
        "minHeight": "100vh",
        "position": "relative",
      }
    }
  >
    <Connect(Navigation) />
    <IndexPageHead />
    <Connect(PageView)>
      <Connect(SearchResults) />
    </Connect(PageView)>
    <Connect(PageView)>
      <Connect(DropZoneComponent) />
    </Connect(PageView)>
  </div>
  <IndexPageDescription />
</div>
`;
